### PR TITLE
Enhance : Admin Notice Show gloablly & extand API

### PR DESCRIPTION
**Pull request by osmansufy**
### All Submissions:

* [ ] My code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)
* [ ] My code satisfies feature requirements
* [ ] My code is tested
* [ ] My code passes the PHPCS tests
* [ ] My code has proper inline documentation
* [ ] I've included related pull request(s) (optional)
* [ ] I've included developer documentation (optional)
* [ ] I've added proper labels to this pull request

### Related Pull Request(s)

* Full PR Link

### Closes 

* Closes #

### How to test the changes in this Pull Request:

* Steps or issue link

### Changelog entry

*Title*

Detailed Description of the pull request. What was previous behaviour
and what will be changed in this PR.

### Before Changes

Describe the issue before changes with screenshots(s).

### After Changes

Describe the issue after changes with screenshot(s).

### Feature Video (optional)

Link of detailed video if this PR is for a feature.

### PR Self Review Checklist:

* Code is not following code style guidelines
* Bad naming: make sure you would understand your code if you read it a few months from now.
* KISS: Keep it simple, Sweetie (not stupid!).
* DRY: Don't Repeat Yourself.
* Code that is not readable: too many nested 'if's are a bad sign.
* Performance issues
* Complicated constructions that need refactoring or comments: code should almost always be self-explanatory.
* Grammar errors.

### FOR PR REVIEWER ONLY:

> As a reviewer, your feedback should be focused on the idea, not the person. Seek to understand, be respectful, and focus on constructive dialog.

> As a contributor, your responsibility is to learn from suggestions and iterate your pull request should it be needed based on feedback. Seek to collaborate and produce the best possible contribution to the greater whole.

* [ ] Correct — Does the change do what it’s supposed to? ie: code 100% fulfilling the requirements?
* [ ] Secure — Would a nefarious party find some way to exploit this change? ie: everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities?
* [ ] Readable — Will your future self be able to understand this change months down the road?
* [ ] Elegant — Does the change fit aesthetically within the overall style and architecture?

## Release Notes

- **New Features**
	- Introduced a new admin notice component to enhance the admin interface.
	- Added a `scope` prop to filter admin notices based on context.
	- Implemented dynamic rendering of admin notices based on the presence of a specific HTML element.
	- Enhanced admin notice handling with improved context management, including a new optional `scope` parameter for REST API requests.
	- Added a global scope key to admin notices for better contextual information.
	- Updated admin notice description for clarity.

- **Bug Fixes**
	- Improved handling of admin notices with enhanced filtering capabilities.

- **Documentation**
	- Updated method signatures and parameters for clarity in admin notice handling.

- **Chores**
	- Expanded Webpack configuration to include the new admin notice entry point.

### CODE REVIEW

File: includes/Admin/Notices/Helper.php [hunk 1]
```diff
@@ -18,13 +18,14 @@ class Helper {
      *
      * @return array | void
      */
-    public static function dokan_get_admin_notices() {
+    public static function dokan_get_admin_notices( $notice_scope = 'local' ) {
```
Comment (mrabbani) [inline includes/Admin/Notices/Helper.php:21]: Pls do not modify this files. Undo all changes.

File: includes/REST/AdminNoticeController.php [hunk 2]
```diff
@@ -36,6 +37,14 @@ public function register_routes() {
                     'methods'             => WP_REST_Server::READABLE,
                     'callback'            => [ $this, 'dokan_get_admin_notices' ],
                     'permission_callback' => [ $this, 'check_permission' ],
+                    'args'                => [
+                        'scope' => [
+                            'description' => __( 'Notice context', 'dokan-lite' ),
+                            'type'        => 'string',
+                            'required'    => false,
+                            'default'     => 'all',
+                        ],
```
Comment (mrabbani) [inline includes/REST/AdminNoticeController.php:46]: Change the default value.

File: includes/REST/AdminNoticeController.php [hunk 3]
```diff
@@ -53,11 +62,13 @@ public function register_routes() {
...
      * Get dokan specific notices
+     * @param WP_REST_Request $request
      *
      * @return WP_REST_Response
      */
-    public function dokan_get_admin_notices() {
-        $notices = Helper::dokan_get_admin_notices();
+    public function dokan_get_admin_notices( WP_REST_Request $request ) {
+        $notice_scope = $request->get_param( 'scope' );
+        $notices        = Helper::dokan_get_admin_notices( $notice_scope );
```
Comment (mrabbani) [inline includes/REST/AdminNoticeController.php:71]: Retrieve all notices and perform filter operation here.

File: src/admin/components/AdminNotice.vue [hunk 4]
```diff
@@ -83,7 +87,7 @@ export default {
     methods: {
         fetch() {
             $.ajax( {
-                url: `${dokan_promo.rest.root}${dokan_promo.rest.version}/admin/notices/${this.endpoint}`,
+                url: `${dokan_promo.rest.root}${dokan_promo.rest.version}/admin/notices/${this.endpoint}?scope=${this.scope}`,
```
Comment (mrabbani) [inline src/admin/components/AdminNotice.vue:90]: Pass `?scope=${this.scope}` only if the `scope` is not empty.

**Review by mrabbani — CHANGES_REQUESTED** [review]

File: includes/REST/AdminNoticeController.php (hunk 2 above)
Comment (mralaminahamed) [reply includes/REST/AdminNoticeController.php:46]: vai, default value changed with your feeback

File: includes/REST/AdminNoticeController.php [hunk 5]
```diff
@@ -53,13 +62,27 @@ public function register_routes() {
...
      */
-    public function dokan_get_admin_notices() {
+    public function dokan_get_admin_notices( WP_REST_Request $request ) {
+        $notice_scope = $request->get_param( 'scope' );
+        $notice_scope = $notice_scope ?? 'local';
+
         $notices = Helper::dokan_get_admin_notices();
+        $notices = array_map(
+            function ( $notice ) {
+                $notice['scope'] = $notice['scope'] ?? 'local';
```
Comment (mrabbani) [inline includes/REST/AdminNoticeController.php:76]:
```php
 function ( $notice ) use($notice_scope)  {
                return  $notice_scope === ($notice['scope'] ?? 'local');
}
```

**Review by mrabbani — CHANGES_REQUESTED** [review]

**Review by mrabbani — APPROVED** [review]

**Comment by mrabbani** [comment]
@getdokan/product, We may need to update the content of the [Dokan data migration notice](https://wedevs.slack.com/archives/D06TB6YEEKB/p1732513938161439).

**Comment by jamil-mahmud** [comment]
@mrabbani vai,

Kindly direct to change the text on Dokan Data Update Required announcement according to below.

"Updating your Dokan data is required to continue functional operations."

**Comment by StalinDurjo** [comment]
Issue: Clicking on upgrader button doesn't work if notice is set to global scope

```
$notices[] = [
            'type'              => 'info',
            'title'             => __( 'Dokan Data Update Required', 'dokan-lite' ),
            'description'       => __( 'We need to update your install to the latest version', 'dokan-lite' ),
            'priority'          => 1,
            'scope'             => 'global',
            'actions'           => [
                [
                    'type'            => 'primary',
                    'text'            => __( 'Update', 'dokan-lite' ),
                    'loading_text'    => __( 'Updating...', 'dokan-lite' ),
                    'competed_text'   => __( 'Updated', 'dokan-lite' ),
                    'reload'          => true,
                    'confirm_message' => __( 'It is strongly recommended that you backup your database before proceeding. Are you sure you wish to run the updater now?', 'dokan-lite' ),
                    'ajax_data'       => [
                        'action'   => 'dokan_do_upgrade',
                        '_wpnonce' => wp_create_nonce( 'dokan_admin' ),
                    ],
                ],
            ],
        ];
``` 

https://jam.dev/c/db9e3cf0-3a63-4bbb-b705-afef35d1ce6a

@mralaminahamed bhai

**Comment by mralaminahamed** [comment]
@StalinDurjo vai, issue fix kore diachi. let check the update

File: src/admin/components/AdminNotice.vue (hunk 4 above)
Comment (mralaminahamed) [reply src/admin/components/AdminNotice.vue]: vai, done.

File: includes/REST/AdminNoticeController.php (hunk 3 above)
Comment (mralaminahamed) [reply includes/REST/AdminNoticeController.php]: vai, done.

File: includes/Admin/Notices/Helper.php (hunk 1 above)
Comment (mralaminahamed) [reply includes/Admin/Notices/Helper.php]: vai, done.

**Pull request merged**